### DISCUSSION TITLE
Reusable workflow action refers to latest release.

### DISCUSSION THREAD

**Discussion by akshit-speed** [community/community#50312] [Actions] [answered]
### Select Topic Area

Question

### Body

We have created couple of reusable workflow in our organization private repositories, we want to access the same from other private repos, which is working good. But as per requirement i want to refer to latest tag (latest release). But there is not any doc which shows that we can perform this or not. I have tried couple of things out of which most suitable i am mentioning below but still it is not working as expected. Is there any work around so we can always refer to latest release in our code. 

**uses: myusername/my-repo/.github/workflows/my-reusable-workflow.yml@latest**
- It is taking latest as a branch name instead of latest release version (tag)
Edit: 
Also tried this syntax. 
`uses: TrySpeed/reusable-workflow/.github/workflows/ios-sonar.yml@{{vars.RELEASE_VERSION}}`
It also gives error, 
- failed to fetch workflow: reference to workflow should be either a valid branch, tag, or commit

**Comment by airtower-luna** [comment, marked as answer]
You can refer to any *Git ref*. So for example you could create a `latest` branch in your repository and when you make a release (ff-)merge it into the branch.

**Reply by akshit-speed** [reply]
Thanks for a suggestion, it is really a good way to ahead.

**Reply by akshit-speed** [reply]
As we don't have any another way to achieve the same, marking this as a best suitable answer.

**Comment by Mohamed-SayedAlAhl** [comment]
To refer to the latest tag of a reusable workflow in GitHub Actions, you can use the following syntax:

```
uses: myusername/my-repo/.github/workflows/my-reusable-workflow.yml@v*
```

The `v*` wildcard will match the latest tag (e.g. `v1.0.0`, `v2.1.3`, etc.) for the specified repository.

Note that this approach only works if you are using tags to version your reusable workflow files. If you are not using tags, you can switch to using tags or create a script that determines the latest version of the workflow file based on other criteria (such as the most recent commit date).

**Reply by akshit-speed** [reply]
Yes we are managing our releases with tags. Will definitely give it a try and check how it goes.

**Reply by akshit-speed** [reply]
`myusername/my-repo/.github/workflows/my-reusable-workflow.yml@v*`

This is giving me error, so we need to find out some another way for same. 
![image](https://user-images.githubusercontent.com/111357442/226316619-60310756-3cfd-4d5a-87e1-b22d7cf548ea.png)

As of now we have only one option that we can go with release/latest branch. But need to figure how can we do same with Tags / versions.

**Reply by Mohamed-SayedAlAhl** [reply]
It seems like the `v*` wildcard syntax for referring to the latest tag is not working for you. In that case, you can try to manually specify the latest tag in the workflow file.

You can find the latest tag by using the following command in your local repository:

```
git describe --tags --abbrev=0
```

This command will output the latest tag in your repository. You can then use this tag in your workflow file like this:

```
uses: myusername/my-repo/.github/workflows/my-reusable-workflow.yml@vX.Y.Z
```

Where `X.Y.Z` is the latest tag in your repository.

By doing this, your workflow will always use the latest version of the reusable workflow file, based on the latest tag in your repository.

**Reply by airtower-luna** [reply]
The `v*` syntax doesn't work because it *doesn't exist*, see the actual syntax documentation for [`jobs.<job_id>.uses`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_iduses).

That "answer" smells of ChatGPT or something similar.

**Reply by akshit-speed** [reply]
Discussed with github support to, they don't support it as of now. I am creating a feature request for the same. 

Also tried this below syntax, but this also not seems to work. 
```
uses: myusername/my-repo/.github/workflows/my-reusable-workflow.yml@{{vars.LATEST_RELEASE}}
```

I have tried above syntax with `organization` variable.
 
So as of now, only way to get the latest data, is to maintain a latest branch as suggested by @airtower-luna.

**Reply by akshit-speed** [reply]
Can one of you create a feature request for the same, i don't able to find a way to create a feature request. @airtower-luna @Mohamed-SayedAlAhl
